index_node/solr.py

```diff
@@ -89,15 +85,16 @@ def start_solr(solr_config_type, port_number, SOLR_INSTALL_DIR="/usr/local/solr"
     start_solr_options = ["start", "-d", server_directory, "-s", solr_solr_home, "-p", port_number, "-a", enable_nodes, "-m", "512m"]
     esg_functions.call_binary("/usr/local/solr/bin/solr", start_solr_options)
 
-    solr_status(SOLR_INSTALL_DIR)
-
-def solr_status(SOLR_INSTALL_DIR):
+def solr_status():
     '''Check the status of solr'''
     try:
-        esg_functions.call_binary("/usr/local/solr/bin/solr", ["status"])
-    except ProcessExecutionError:
-        logger.error("Error checking solr status")
+        result = esg_functions.call_binary("/usr/local/solr/bin/solr", ["status"])
+    except ProcessExecutionError as error:
+        # 3 is the return code if not running
+        if error.retcode==3:
+            return False
         raise
+    return True
 
 #TODO: fix and test
 def check_solr_process(solr_config_type="master", port=8984):
```

Comment:
Does your fix return the status for all solr processes?  I know I had this TODO where I was going to check both of the processes that got started (on ports 8984 and 8983)

Reply:
It will return True if *any* `solr` services are running I believe. So we lose the ability to detect if *both* came up, we can only see if there is any up. The old function wasn't working properly and was saying there was none up. Now at least valid information is provided.

Reply:
I still have another change on this, so hold up on merging it.

Reply:
I am done making changes on this branch, unless we would like to make a check for both ports to be more thorough.